tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/common/util/DivviupUtils.kt

```diff
@@ -34,7 +43,302 @@ object DivviupUtils {
+                val helperEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_helper))
+                val timePrecisionSeconds: Long =
+                    context.resources.getInteger(R.integer.divviup_count_unlocks_timePrecisionSeconds)
+                        .toLong()
+                val client = Client.createPrio3Count(
+                    context, leaderEndpoint, helperEndpoint, taskId, timePrecisionSeconds
+                )
+                client.sendMeasurement(true)
+                CommonPreferences.setInstallMetricSent(true)
+                Timber.d("Divviup runInstallEvent measurement sent")
+            } catch (e: Exception) {
+                Timber.e(e, "Divviup sending runInstallEvent failed")
+            }
+        }
+    }
+
+    /**
+     * This task is called on Lock
+     * and sending time spent since unlock
+     **/
+    fun runTimeSpentEvent(context: Context, milliseconds: Long) {
+        Executors.newSingleThreadExecutor().execute {
+            try {
+                val taskId = TaskId.parse(context.getString(R.string.divviup_time_spent_id))
+                val leaderEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_leader))
+                val helperEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_helper))
+                val timePrecisionSeconds: Long =
+                    context.resources.getInteger(R.integer.divviup_count_unlocks_timePrecisionSeconds)
+                        .toLong()
+                val client = Client.createPrio3Sum(
+                    context, leaderEndpoint, helperEndpoint, taskId, timePrecisionSeconds, 32
+                )
+                client.sendMeasurement(milliseconds)
+                Timber.d("Divviup runTimeSpentEvent measurement sent %s", milliseconds)
+            } catch (e: Exception) {
+                Timber.e(e, "Divviup sending runTimeSpentEvent failed")
+            }
+        }
+    }
+
+    /**
+     * This task is called when photo is taken
+     * using Camera
+     **/
+    fun runPhotoTakenEvent(context: Context) {
+        if (!CommonPreferences.hasAcceptedAnalytics() || isDebugBuild(context)) return
+        Executors.newSingleThreadExecutor().execute {
+            try {
+                val taskId = TaskId.parse(context.getString(R.string.divviup_count_photos_id))
+                val leaderEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_leader))
+                val helperEndpoint: URI =
+                    URI.create(context.getString(R.string.divviup_helper))
+                val timePrecisionSeconds: Long =
+                    context.resources.getInteger(R.integer.divviup_count_unlocks_timePrecisionSeconds)
+                        .toLong()
+                val client = Client.createPrio3Count(
+                    context, leaderEndpoint, helperEndpoint, taskId, timePrecisionSeconds
+                )
```

Comment:
Also to get rid of the duplicated code we can create a Generic method to handle event execution with error handling, something like this -> 
 ```
   private fun runEvent(
        shouldRun: Boolean,
        updatePreferences: () -> Unit = {},
        eventAction: () -> Unit
    ) {
        if (!shouldRun) return

        scheduler.scheduleDirect {
            try {
                eventAction()
                updatePreferences()
                Timber.d("Divviup event measurement sent")
            } catch (e: Exception) {
                Timber.e(e, "Divviup event measurement sending failed")
            }
        }
    }
```